### ISSUE TITLE
Provide a way to easily find correct versions alignements

### ISSUE THREAD

**Issue by ia3andy** [openrewrite/rewrite#3367]
## What problem are you trying to solve?
As explained on zulip, https://rewriteoss.slack.com/archives/C01AB6L98TC/p1687767750280069:

W have an [artifact](https://github.com/quarkusio/quarkus-updates/tree/main/recipes) with dependency on various OpenRewrite modules, this also reference Maven and Gradle plugins version in a [property](https://github.com/quarkusio/quarkus-updates/blob/main/recipes/src/main/resources/quarkus-updates/recipes.properties) file (to be use to run the [yaml](https://github.com/quarkusio/quarkus-updates/blob/main/recipes/src/main/resources/quarkus-updates/core/3alpha.yaml)/[java](https://github.com/quarkusio/quarkus-updates/blob/main/recipes/src/main/java/io/quarkus/updates/core/quarkus30/AdjustKotlinAllOpenDirectives.java) recipes  in that artifact). Then I want to use that artifact from either the Maven or Gradle build plugin using the version of the plugins specified in the properties. but we need to know which versions of the plugins are compatible with the modules used in your artifact

## Describe the solution you'd like

Have the Gradle/Maven Rewrite plugins versions aligned with the OpenRewrite core version (at least up to the minor).

## Have you considered any alternatives or workarounds?

Yes but there is not a lot of other solutions

## Additional context
Link are provided in the description

## Are you interested in [contributing this feature to OpenRewrite]

Sure

**Comment by timtebeek** [comment]
After some discussion internally we came to the conclusion that the Maven and Gradle plugins version numbers are likely to diverge soon after any alignment, as we are use to develop minor features to build plugins independent of minor versions of OpenRewrite. That for now rules out aligning the versions of the plugins with OpenRewrite, unfortunately.

That leaves open the option to better document the compatible versions of the plugins and OpenRewrite. @mike-solomon is there any support for instance to add tags to openrewrite/rewrite-docs with each new release? Perhaps also going back to the latest 7.x? That way we'd have a snapshot of the different versions of this particular page: https://docs.openrewrite.org/reference/latest-versions-of-every-openrewrite-module
[latest-versions-of-every-openrewrite-module.md](https://github.com/openrewrite/rewrite-docs/blob/master/reference/latest-versions-of-every-openrewrite-module.md)
Maybe there's even a way in GitBook to get a version number dropdown going such that we can also document historic versions?
That could help to have an easier way to access what historic versions are compatible, since we're not cross-compatible over time.

Open to other suggestions as well; we already capture [the rewrite.version](https://github.com/openrewrite/rewrite-maven-plugin/blob/530150dad313c0929612fc9c1e0f8c179bb14589/pom.xml#L68) with each Maven plugin release for instance; could that be of some use here @ia3andy ?

**Comment by mike-solomon** [comment]
@timtebeek I don't believe that GitBook supports versioning like that @timtebeek :( - I was looking at Antora with AsciiDoc [over here](https://github.com/openrewrite/rewrite-docs/issues/162) and they both support it if we ever migrate to that.

I think the only two options we really have with GitBook are:

1. Make a new page for every release (similar to the changelog). We could perhaps add the latest versions to the bottom of the existing changelogs.
2. Make one giant page that has all of the OpenRewrite versions with the table duplicated/changed for each.

Using tabs wouldn't be feasible as they get really difficult to view or cycle through once there's more than a few. There aren't any other structures that GitBook offers for grouping as far as I can see :(
